### PR TITLE
Add event for keyboard close

### PR DESCRIPTION
**Pull request by Hekku2**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | #12
| License       | MIT

- Close event is now emitted after keyboard is closed

This can be tested with https://www.npmjs.com/package/@protacon/ng-virtual-keyboard/v/0.10.0-beta.1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ Simple usage example
   ng-virtual-keyboard
   ng-virtual-keyboard-layout="alphanumeric"
   ng-virtual-keyboard-placeholder="Custom placeholder text"
+  (ng-virtual-keyboard-placeholder)="onClose()"
```
Comment (tarlepp) [inline README.md:42]: Should this be `ng-virtual-keyboard-close` ?

File: app/app.component.html [hunk 2]
```diff
@@ -56,10 +56,10 @@ <h2>Usage</h2>
   ng-virtual-keyboard
   ng-virtual-keyboard-layout="alphanumeric"
   ng-virtual-keyboard-placeholder="Custom placeholder text"
+  (ng-virtual-keyboard-placeholder)="onOpen()"
```
Comment (tarlepp) [inline app/app.component.html:59]: Same here as above - also method name is opposite one.

File: src/virtual-keyboard.directive.ts [hunk 3]
```diff
@@ -75,6 +76,7 @@ export class NgVirtualKeyboardDirective {
       dialogRef
         .afterClosed()
         .subscribe(() => {
+          this.keyboardClose.emit(null);
```
Comment (tarlepp) [inline src/virtual-keyboard.directive.ts:79]: Could this be just `.emit()` ?

File: dist/virtual-keyboard.directive.d.ts [hunk 4]
```diff
@@ -9,6 +9,7 @@ export declare class NgVirtualKeyboardDirective {
     layout: KeyboardLayout | string;
     placeholder: string;
     type: string;
+    keyboardClose: EventEmitter<any>;
```
Comment (tarlepp) [inline dist/virtual-keyboard.directive.d.ts:12]: Do we need that `<any>` here?

**Review by tarlepp — CHANGES_REQUESTED** [review]

File: src/virtual-keyboard.directive.ts (hunk 3 above)
Comment (Hekku2) [reply src/virtual-keyboard.directive.ts:79]: Good point!

File: README.md (hunk 1 above)
Comment (Hekku2) [reply README.md:42]: Damn these copy paste devils.

File: dist/virtual-keyboard.directive.d.ts (hunk 4 above)
Comment (Hekku2) [reply dist/virtual-keyboard.directive.d.ts:12]: Perhaps `void` is better here, EventEmitter type definition seems to require some type.

Comment (tarlepp) [reply dist/virtual-keyboard.directive.d.ts]: yeah, `void` will tell that there isn't anything to emitted on that event - it's just an event, which makes totally sense in this context - imho

File: src/virtual-keyboard.directive.ts [hunk 5]
```diff
@@ -20,9 +20,10 @@ export class NgVirtualKeyboardDirective {
   private opened = false;
   private focus = true;
 
-  @Input('ng-virtual-keyboard-layout') layout: KeyboardLayout|string;
+  @Input('ng-virtual-keyboard-layout') layout: KeyboardLayout | string;
   @Input('ng-virtual-keyboard-placeholder') placeholder: string;
   @Input('ng-virtual-keyboard-type') type: string;
+  @Output('ng-virtual-keyboard-close') keyboardClose: EventEmitter<void> = new EventEmitter();
```
Comment (tarlepp) [inline src/virtual-keyboard.directive.ts:26]: Personally I prefer to initialize these on `construct` method - but making this like this isn't really a problem just personal preference.

**Review by tarlepp — APPROVED** [review]
Looks good, just one small comment.

**Pull request merged**